x-pack/plugins/observability/public/components/shared/header_menu_portal.test.tsx

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { render } from '@testing-library/react';
+import React from 'react';
+import HeaderMenuPortal from './header_menu_portal';
+
+describe('HeaderMenuPortal', () => {
+  describe('render', () => {
+    it('sets the header action menu', () => {
+      const setHeaderActionMenu = jest.fn();
+
+      render(<HeaderMenuPortal setHeaderActionMenu={setHeaderActionMenu}>test</HeaderMenuPortal>);
+
+      expect(setHeaderActionMenu).toHaveBeenCalled();
```

Comment:
Would be great if you can also verify that `setHeaderActionMenu` is called with `undefined` when it unmounts.